src/pulse3D/excel_writer.py

```diff
@@ -221,6 +225,61 @@ def write_xlsx(
     interpolated_data_period = (
         w[INTERPOLATION_VALUE_UUID] if plate_recording.is_optical_recording else INTERPOLATED_DATA_PERIOD_US
     )
+
+    # get stim meta data
+    if include_stim_protocols:
+        # no stimulation used
+        if w[STIMULATION_PROTOCOL_UUID] == "null":
```

Comment:
This only checks the first file (Well A1), so we need to check every well before setting the DF to this value. Also your loop down below is already set up to check each well to see if it has a protocol assigned to it. You could check `unassigned_wells` after looping through each well to see if no wells have protocols assigned. In that case, you'd want to make it a list so you can just check `no_protocol_assigned_on_any_wells = len(unassigned_wells) == 24`. That's just one way of doing though